### DISCUSSION TITLE
I get an undefined name on a name-change.

### DISCUSSION THREAD

**Discussion by pstuifzand** [davidfig/tree#6] [General]
It seems to happen here. The `name` variable is the new text, because the `this._input.value` is `undefined`.

Do I use it wrong, or is this a bug?

https://github.com/davidfig/tree/blob/28571764fff6bf73e1db2c133b1cc24ca5365ab1/src/tree.js#LL360C8-L360C9

**Comment by davidfig** [comment]
How is `this._input` undefined? It's defined in the constructor and not deleted? Is it a closure issue?

**Reply by pstuifzand** [reply]
I use it like this: https://github.com/pstuifzand/crdt-tree-algorithm/blob/master/src/index.js#L18-L22

The `value` is undefined. So the name in the emitted event is undefined. As a workaround I use leaf.data.name instead of the argument that is passed.
